docs/adr/0030-disable-squash-and-merge.md

```diff
@@ -0,0 +1,47 @@
+
+GitHub sets user's default merge strategy to whatever they used last (squash and merge or regular merge) and doesn't allow disabling the option on certain branches. These caveats, alongside human error and automated dependency upgrades, have caused a number of problems we've identified:
+
+1. Accidentally squashing and merging train branches into `main` causes merge conflicts for the next patch release because of differing parent commits
+1. Accidentally forgetting to edit the commit message when squashing and merging an approved PR means PRs can be merged without meeting our commit guidelines
+1. Accidentally regular merging a branch that was meant to be squashed and merged means multiple unformatted commits can land for a single task
+1. While squashing and merging Dependabot (`package.json`) and Bananafox (`yarn.lock`) commits on automated dependency PRs seemed like a nice _pro_ for keeping squash and merge, there's been a couple of instances where something funky happened with a bad squash and merge dependency update which caused Bananafox to get out of sync and commit on multiple PRs without `package.json` changes. This caused an issue with a dot release that could have been mitigated by being able to cherry-pick in only the needed change without the `yarn.lock` commit, see [this PR](https://github.com/mozilla/fxa/pull/12064) and and others around this time, and so it seems more ideal to keep these in separate commits until Dependabot [works properly with yarn 2](https://github.com/dependabot/dependabot-core/issues/1297) so we can retire Bananafox
+
+## Considered Options
+
+- Continue allowing squash and merge through GH
+- Disable squash and merge
+
+## Decision Outcome
+
+Chosen option: "option 2", because with our current workflow, team discussions have concluded there doesn't appear to be adequate workarounds to alleviate the problems listed above while leaving squash and merge enabled. This is not a final decision; we will continue to have conversations around development strategy process improvements and viability of squash and merge in the future.
```

Comment:
Should we also decide to officially disable force push on train-* branches too? Seeing the comments and suggestions below, this seems like a good call.

Reply:
This WFM 👍